jina/helper.py

```diff
@@ -693,12 +696,16 @@ def run(self):
         return asyncio.run(func(*args, **kwargs))
 
 
-def check_keys_exist(keys_to_check, existing_keys):
-    missed = []
-    for k in keys_to_check:
-        if k not in existing_keys:
-            missed.append(k)
-    return missed
+def filter_keys_nonexistent(keys: Iterator, keys_existing: Iterator, check_path: str, logger: 'JinaLogger'):
```

Comment:
`keys_existing` should not be type of Iterator but maybe Set.

Reply:
It depends on how the indexer stores the keys. I will cast them to `set` to be sure during the calculation

Reply:
if this is to be used from different points then have a private _delete with an extra argument and have it called from update and delete method with the proper value.

Reply:
Sorry is not in this discussion but innthe kwargs one